### PR TITLE
Internal: Internal: Add 'Free to Pro' Motion Effects promotion [ED-13458]

### PR DESCRIPTION
**Pull request by davseve**

### CODE REVIEW

File: modules/promotions/assets/js/editor/module.js [hunk 1]
```diff
@@ -21,7 +21,8 @@ export default class Module extends elementorModules.editor.utils.Module {
 	}
 
 	registerControlBehavior( behaviors, view ) {
-		if ( 'display_conditions_pro' !== view.options.model.get( 'name' ) ) {
+		const controlsToSkip = [ 'display_conditions_pro', 'scrolling_effects_pro', 'mouse_effects_pro', 'sticky_pro' ];
+		if ( ! controlsToSkip.includes( view.options.model.get( 'name' ) ) ) {
```
Comment (louiswol94) [inline modules/promotions/assets/js/editor/module.js:25]:
```suggestion
		const promotionsControls = [ 'display_conditions_pro', 'scrolling_effects_pro', 'mouse_effects_pro', 'sticky_pro' ];
		if ( ! promotionsControls.includes( view.options.model.get( 'name' ) ) ) {
```
Maybe something like this? It is not controls to skip, it's controls that we want to run the logic for.

File: includes/managers/controls.php [hunk 2]
```diff
@@ -1257,4 +1257,106 @@ public function add_display_conditions_controls( Controls_Stack $controls_stack
...
+							<span class="elementor-switch-label" data-off="Off"></span>
+							<span class="elementor-switch-handle"></span>
+						</label>
+					</div>
+				</div>
+			</div>
+		</div>';
+	}
+
+	private function promotion_control_mouse_effects(): string {
```
[the remark this answers is not in the document]
Comment (davseve) [reply includes/managers/controls.php:1323]: Actually, they are identical besides the title text.
I thought about that, the reason I didn't do one function is that the translation function must receive a string, the CI will fail if I try to pass a variable.
But I am open to suggestions.

[the remark this answers is not in the document]
Comment (mykytamurzin) [reply includes/managers/controls.php:1323]: can we pass it in such way ? And within function just do a regular echo or `Utils::print_unescaped_internal_string`
```
function foo( esc_html__( 'Scrolling Effects', 'elementor' ) ) { 
```

[the remark this answers is not in the document]
Comment (mykytamurzin) [reply includes/managers/controls.php:1323]: I'm looking into `get_teaser_template` function

[the remark this answers is not in the document]
Comment (davseve) [reply includes/managers/controls.php:1323]: That exactly what I did :)

[the remark this answers is not in the document]
Comment (louiswol94) [reply includes/managers/controls.php:1323]: https://github.com/elementor/elementor/blob/main/includes/widgets/video.php#L959

[the remark this answers is not in the document]
Comment (davseve) [reply includes/managers/controls.php]: The var is in the escaping, not in the translation :)

File: modules/promotions/assets/js/editor/module.js (hunk 1 above)
Comment (davseve) [reply modules/promotions/assets/js/editor/module.js]: Fixed

**Review by murzin-nikita — DISMISSED** [review]

File: tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts [hunk 3]
```diff
@@ -1,26 +1,61 @@
-import { test, expect, type Page } from '@playwright/test';
+// @ts-ignore
```
Comment (dmaziar) [inline tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts:1]: why do we need this here?

File: tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts [hunk 4]
```diff
@@ -1,26 +1,61 @@
...
+		const wpAdminPage = new WpAdminPage( page, testInfo ),
+			promotionContainer = '.e-feature-promotion';
 
-	await wpAdminPage.login();
+		await wpAdminPage.login();
 
-	await test.step( 'Free to Pro - Submissions', async () => {
-		await wpAdminPage.promotionPageScreenshotTest( promotionContainer, 'e-form-submissions', 'submissions-menu-item-desktop' );
-	} );
+		await test.step( 'Free to Pro - Submissions', async () => {
```
Comment (dmaziar) [inline tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts:13]: I'm not a fan of this `test.step`. If you are using page object and wrap playwright code in functions, then it is readable and makes no sense in additional wrappers

Comment (dmaziar) [reply tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts:13]: But this is up to you to leave it or remove

Comment (davseve) [reply tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts:13]: I think I will leave it, just to increase the readability

File: tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts (hunk 3 above)
Comment (davseve) [reply tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts]: Removed

File: tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts [hunk 5]
```diff
@@ -1,26 +1,61 @@
...
+			const promotionControls = [ 'elementor-control-scrolling_effects_pro', 'elementor-control-mouse_effects_pro', 'elementor-control-sticky_pro' ];
+
+			for ( const control of promotionControls ) {
+				const controlContainer = page.locator( `.${ control }` );
+				await expect.soft( controlContainer ).toHaveScreenshot( `${ control }.png` );
+			}
+		} );
+
+		await test.step( 'Free to Pro - Scrolling Effect Modal', async () => {
+			await editor.promotionModalScreenshotTest( 'scrolling-effects' );
```
[the remark this answers is not in the document]
Comment (davseve) [reply tests/playwright/sanity/modules/promotions/admin-menu-items/menu-item-promotions.test.ts]: Fixed :)

File: tests/playwright/sanity/modules/promotions/helper.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { type Page, expect } from '@playwright/test';
+
```
Comment (dmaziar) [inline tests/playwright/sanity/modules/promotions/helper.ts:2]:
```
readonly page: Page;
	readonly editorPage: EditorPage;
	constructor( page: Page, testInfo: TestInfo ) {
		this.page = page;
		this.editorPage = new EditorPage( this.page, testInfo );
	}
```

Comment (dmaziar) [reply tests/playwright/sanity/modules/promotions/helper.ts:2]: Can you add a constructor and then use this.page instead of passing it as a params?

**Review by dmaziar — APPROVED** [review]

**Pull request merged**